liam2/alignment.py

```diff
@@ -115,11 +115,21 @@ def align_get_indices_nd(ctx_length, groups, need, filter_value, score,
                 else:
                     group_maybe_indices = members_indices
                 if isinstance(score, np.ndarray):
-                    maybe_members_rank_value = score[group_maybe_indices]
-                    # TODO: use np.partition (np1.8+)
-                    sorted_local_indices = np.argsort(maybe_members_rank_value)
-                    sorted_global_indices = \
-                        group_maybe_indices[sorted_local_indices]
+                    if method == 'default':
+                        maybe_members_rank_value = score[group_maybe_indices]
+                        # TODO: use np.partition (np1.8+)
+                        sorted_local_indices = np.argsort(maybe_members_rank_value)
+                        sorted_global_indices = \
+                            group_maybe_indices[sorted_local_indices]
+                    if method == 'sidewalk':
+                        score_max = max(score[group_maybe_indices])
+                        score_min = min(score[group_maybe_indices])

```

Comment:
It would probably be better to do the check on the whole score vector, not on each group.